plugins/modules/ndo_pod_settings.py

```diff
@@ -0,0 +1,304 @@
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template_name = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    macsec_policy = module.params.get("macsec_policy")
+    macsec_policy_uuid = module.params.get("macsec_policy_uuid")
+    ntp_policy = module.params.get("ntp_policy")
+    ntp_policy_uuid = module.params.get("ntp_policy_uuid")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template_name, template_id)
+    mso_template.validate_template("fabricPolicy")
+    object_description = "Pod Settings"
+
+    match = mso_template.get_pod_settings_object(uuid, name)
+
+    if (uuid or name) and match:
+        set_pod_settings_policy_names(mso_template, match.details)
+        mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = [set_pod_settings_policy_names(mso_template, obj) for obj in match]  # Query all objects
+
+    pod_settings_path = "/fabricPolicyTemplate/template/podPolicyGroups/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+
+        mso_values = {
+            "name": name,
+            "description": description,
+        }
+
+        if macsec_policy or macsec_policy_uuid:
+            mso_values["fabricMACsec"] = mso_template.get_macsec_policy_object(macsec_policy_uuid, macsec_policy, fail_module=True).details.get("uuid")
+        if ntp_policy or ntp_policy_uuid:
+            mso_values["ntp"] = mso_template.get_ntp_policy_object(ntp_policy_uuid, ntp_policy, fail_module=True).details.get("uuid")
+
+        if match:
+            remove_data = []
+            unwanted = []
+            if macsec_policy == "" or macsec_policy_uuid == "":
+                remove_data.append("fabricMACsec")
+                unwanted.extend(["fabricMACsec", "macsec_policy"])
+            if ntp_policy == "" or ntp_policy_uuid == "":
+                remove_data.append("ntp")
+                unwanted.extend(["ntp", "ntp_policy"])
+
+            append_update_ops_data(ops, match.details, pod_settings_path, mso_values, remove_data)
+            mso.sanitize(mso_values, collate=True, unwanted=unwanted)
+            set_pod_settings_policy_names(mso_template, mso.proposed)
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=pod_settings_path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=pod_settings_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = mso_template.get_pod_settings_object(uuid, name, search_object=response)
+        if match:
+            set_pod_settings_policy_names(mso_template, match.details)
```

Comment:
->
mso_template.update_config_with_template_and_references(match.details, reference_dict)

Reply:
resolving, will replace function call if the discussions lead to changing the function